### DISCUSSION TITLE
脑图是否存在某个属性可以判断当前节点是在根节点左边还是右边？

### DISCUSSION THREAD

**Discussion by tuo-gu** [antvis/G6#3755] [Ideas]
**前景：**
layout: {
        type: 'mindmap',
        direction: 'H',
        getSide: (d) => {
          if (d.id === 'Classification') {
            return 'left';
          }
          return 'right';
        }
      }
已知在代码中可以通过设置 return 'left';让id=Classification的节点及其子节点成为左布局。
需求是：根节点存在上游和下游且都支持缩展。
现在的想法是下游可以用children+collapsed控制缩展，上游的话（因为G6树图只能有一个父节点）可能存在不止一个父节点只能用一个children放到左边模拟父节点。

**问题：**左布局的节点样式跟右布局的一致了，右边布局本身流转方向是从左到右，折叠按钮放在最右边没问题。左边布局本身流转方向是从右到左，折叠按钮也放在最右边了。这个就不符合需求了，请问有无属性判断该节点是否属于左布局？从而可以自定义节点的时候做个判断调整一下

**Comment by zyh0yh** [comment]
可以使用 获取当前的节点source节点来判断，判断source节点的 x 与当前节点的 x 来判断是左布局还是右布局
